precompiles/utils/src/tests.rs

```diff
@@ -0,0 +1,442 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+use sp_core::{H256, U256};
+
+fn u256_repeat_byte(byte: u8) -> U256 {
+	let value = H256::repeat_byte(byte);
+
+	U256::from_big_endian(value.as_bytes())
+}
+
+#[test]
+fn write_u256() {
+	let value = U256::from(42);
+
+	let output = EvmDataWriter::new().write(value).build();
+
+	let mut expected_output = [0u8; 32];
+	value.to_big_endian(&mut expected_output);
+
+	assert_eq!(output, expected_output);
+}
+
+#[test]
+fn read_u256() {
+	let value = U256::from(42);
+	let writer_output = EvmDataWriter::new().write(value).build();
+
+	let mut reader = EvmDataReader::new(&writer_output);
+	let parsed: U256 = reader.read().expect("to correctly parse U256");
+	reader.check_complete().expect("complete");
+
+	assert_eq!(value, parsed);
+}
+
+#[test]
+#[should_panic(expected = "to correctly parse U256")]
+fn read_u256_too_short() {
+	let value = U256::from(42);
+	let writer_output = EvmDataWriter::new().write(value).build();
+
+	let mut reader = EvmDataReader::new(&writer_output[0..31]);
+	let _: U256 = reader.read().expect("to correctly parse U256");
```

Comment:
It feels weird to call `.expect` and also use `should_panic`. Why not just assert that the result is the correct `Err` value?

Reply:
It seems shorter and simpler than doing an `if let`/`match` and panicking.